src/WebJobs.Script/OutOfProc/FunctionDispatcherFactory.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+using Microsoft.Azure.WebJobs.Script.Eventing;
+using Microsoft.Azure.WebJobs.Script.ManagedDependencies;
+using Microsoft.Azure.WebJobs.Script.OutOfProc.Http;
+using Microsoft.Azure.WebJobs.Script.Rpc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.OutOfProc
+{
+    internal class FunctionDispatcherFactory : IFunctionDispatcherFactory
+    {
+        private IFunctionDispatcher _functionDispatcher;
+
+        public FunctionDispatcherFactory(IOptions<ScriptJobHostOptions> scriptHostOptions,
+            IMetricsLogger metricsLogger,
+            IScriptJobHostEnvironment scriptJobHostEnvironment,
+            IScriptEventManager eventManager,
+            ILoggerFactory loggerFactory,
+            IHttpWorkerChannelFactory httpWorkerChannelFactory,
```

Comment:
The fact this is working with distinct abstractions for HTTP and RPC workers is an issue. Is there a reason we can't have common abstractions here?

Reply:
Rpc and Http dispatchers have different dependencies:
- LanguageWorkerOptions vs HttpWorkerOptions
- Rpc workers have placeholder channels 

Let me know if you have ideas on how to unify the dependencies

Reply:
Opened https://github.com/Azure/azure-functions-host/issues/5126 follow up task